core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/codeGen/CodeWithConverter.kt

```diff
@@ -7,14 +7,19 @@ import org.jetbrains.kotlinx.jupyter.api.VariableName
  * Class representing generated code declarations for a [Marker].
  *
  * @param declarations The generated code.
+ * @param converter Needs to provide additional info (name) from org.jetbrains.dataframe.impl.codeGen.CodeGenerator to its callers
+ * But at the same time name doesn't make sense for GroupBy where code to be executed contains two declarations
  * @param converter Optional converter for the [Marker], such as a [org.jetbrains.kotlinx.dataframe.api.cast], often used for Jupyter.
  */
-public data class CodeWithConverter(val declarations: Code, val converter: (VariableName) -> Code = EmptyConverter) {
+public data class CodeWithConverter<T : CodeConverter>(
+    val declarations: Code,
+    val converter: T
```

Comment:
maybe it's an idea to rename this variable or rework it a bit, because looking at this with new eyes, even with the documentation, I have no idea what's going on

Reply:
Idk, hard to tell more than expressed in the code itself. Basically CodeWithConverter is all the information that dataframe integration needs to generate except name of the variable being updated. Variable name is provided by the caller due to separation of concerns